core/include/connection.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
```

Comment:
The problem with moving `Connection` to the public header files is that we expose the `Connection` base class which doesn't need to be public. Also, this means that `mavlink_receiver.h` needs to be included because symbols of it are required in `connection.h`, however, `mavlink_receiver.h` is not public either.

So this means that we need to have some separation for this and I believe that's why we had the connection result defined in `dronecode_sdk.h` previously (and then later `connection_result.h`).

Reply:
I fixed all the other small problems also the one with the indentations.
Thanks for pointing to me this issue, I understand that users should not be able to use the Connection class directly. Thus, we can move it to the core directory. 
Are you ok with this modification??
I am going to send you a pull request with the above modifications.

Reply:
@JonasVautherin I'm sure you have an opinion about that? :smile:

Reply:
Haha, I was the one making `connection_result.h` in the first place, so I am not sure if I have a better solution :sweat_smile:. I had two reasons for that:

1. A `ConnectionResult` is a status, and not an internal object of `Connection`. We expose statuses, and not `Connection`, so I was in favor of not having them in the same header.
2. When mocking (e.g. [here](https://github.com/Dronecode/DronecodeSDK/blob/develop/core/mocks/dronecode_sdk_mock.h)), I don't need the `Connection` header, precisely because only the statuses are exposed. So it was looking cleaner to me to not have the full header there as well.

This said, I don't have a super strong opinion about that, and I am open to other solutions. But I can't say I have a better idea :/.

Reply:
I will try to find a solution in the next few days. Now, I need to find a way to reset my EKF2
during the simulation as we have discussed in the forum :smiley: